src/main/scala/com/fullcontact/interview/RecordFinder.scala

```diff
@@ -1,7 +1,71 @@
 package com.fullcontact.interview
 
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.{DataFrame, SaveMode, SparkSession}
+
 object RecordFinder {
+  val Identifier = "identifier"
+  val Query = "query"
+  val Record = "record"
+  val Records = "records"
+
   def main(args: Array[String]): Unit = {
-    // TODO: Implement your job here
+    val spark = SparkSession.builder()
+      .master("local[4]")
+      .getOrCreate()
+
+    val records = spark.read.csv("./Records.txt").toDF(Record).repartition(4)
+    val queries = spark.read.csv("./Queries.txt").toDF(Query).repartition(4)
+
+    val recordsById = explodeRecordsDF(records)
+
+    val withDupes = recordsById.join(queries, recordsById(Identifier) === queries(Query))
+      .select(queries(Query), recordsById(Record))
+
+    import spark.sqlContext.implicits._
+    withDupes
+      .map(row => {
+        s"${row.getAs(Query)}: ${row.getAs(Record)}"
+      })
+      .coalesce(1)
```

Comment:
How did you choose to use `coalesce` here, and how would the data growing in size influence your decision?

How does this `coalesce` interact with the upstream `repartition`?